apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -131,8 +132,15 @@ class PageTemplateModal extends Component {
 		if ( event.target.matches( 'button.template-selector-item__label' ) ) {
 			return false;
 		}
-		this.setState( { isOpen: false } );
+
+		this.setState( { wasClosed: true } );
 		trackDismiss( this.props.segment.id, this.props.vertical.id );
+
+		if ( window && window.history && window.history.length ) {
+			window.history.back();
```

Comment:
Just wondering if `constructor` is where we should be handling such things? I wonder whether the life cycle methods would be more appropriate?

Reply:
I'm not sure if I understand the comment, here. The idea is just to get back to the previous page when the user clicks on the `X` button. Thinking in the wp-admin, where usually the user can start to add a new through of the `Add Page` submenu:

![image](https://user-images.githubusercontent.com/77539/66078392-50522200-e538-11e9-8b60-8e7500deacba.png)

So if the current page is the users, for instance as above, when the user clicks on the `X` button, it should show the users page again.

Reply:
@apeatling @shaunandrews What are your thoughts here?

Reply:
Reading again the issue I see now that the idea is/was get the page lists. By mistake, I understood a different behavior, which maybe is a feature? Sorry about this confusion.

Reply:
This comment was specifically about using the `constructor` for side effects.

Note this from [the React docs](https://reactjs.org/docs/react-component.html#constructor):

> Avoid introducing any side-effects or subscriptions in the constructor. For those use cases, use componentDidMount() instead.

I'm flagging `window.history.back()` as a side effect that would be better handled in a lifecycle method.